src/setup/sound.c

```diff
@@ -142,26 +149,37 @@ static void UpdateExtraTable(TXT_UNCAST_ARG(widget),
     // Rebuild the GUS table. Start by emptying it, then only add the
     // GUS control widget if we are in GUS music mode.
 
-    TXT_ClearTable(extra_table);
-
-    if (snd_musicmode == MUSICMODE_GUS)
+    if (snd_musicmode == MUSICMODE_OPL)
```

Comment:
Change to switch (snd_musicmode) and show different things depending on music type.